tests/Unit/Domain/Test_CreateInitialElement.cpp

```diff
@@ -21,37 +28,222 @@
 #include "Domain/Neighbors.hpp"
 #include "Domain/OrientationMap.hpp"
 #include "Domain/SegmentId.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
 #include "Utilities/MakeArray.hpp"
+#include "Utilities/StdHelpers.hpp"
 
 namespace {
 void test_create_initial_element(
     const ElementId<2>& element_id, const Block<2>& block,
+    const std::vector<std::array<size_t, 2>>& refinement_levels,
     const DirectionMap<2, Neighbors<2>>& expected_neighbors) noexcept {
-  const auto created_element =
-      domain::Initialization::create_initial_element(element_id, block);
+  const auto created_element = domain::Initialization::create_initial_element(
+      element_id, block, refinement_levels);
   const Element<2> expected_element{element_id, expected_neighbors};
   CHECK(created_element == expected_element);
 }
+
+void test_h_refinement() noexcept {
+  const auto check_neighbors =
+      [](const OrientationMap<3>& neighbor_orientation,
+         const std::array<size_t, 3>& neighbor_refinement,
+         const std::unordered_set<std::array<SegmentId, 3>,
+                                  boost::hash<std::array<SegmentId, 3>>>&
+             expected_upper_neighbors) noexcept {
+    CAPTURE(neighbor_orientation);
+    CAPTURE(neighbor_refinement);
+    for (bool upper_face : {true, false}) {
+      CAPTURE(upper_face);
+      const auto test_direction =
+          upper_face ? Direction<3>::upper_xi() : Direction<3>::lower_xi();
+      const ElementId<3> self_id(
+          0, {{{1, upper_face ? 1_st : 0_st}, {1, 0}, {1, 1}}});
+      const Block<3> self_block(
+          domain::make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
+              domain::CoordinateMaps::Identity<3>{}),
+          0, {{test_direction, {1, neighbor_orientation}}});
+      const std::vector<std::array<size_t, 3>> refinement_levels{
+          {{1, 1, 1}}, neighbor_refinement};
+      std::unordered_set<ElementId<3>> expected_neighbors;
+      for (auto segment_ids : expected_upper_neighbors) {
+        if (not upper_face) {
+          const auto neighbor_perpendicular = neighbor_orientation(0);
+          gsl::at(segment_ids, neighbor_perpendicular) =
+              gsl::at(segment_ids, neighbor_perpendicular).id_if_flipped();
+        }
+        expected_neighbors.insert({1, segment_ids});
+      }
```

Comment:
instead of computing the `expected_neighbors`, can you just pass the expectation to the lambda? tests where the expected value is computed in a way that's not immediately obvious always ring alarm bells for me..

Reply:
I was trying to avoid writing out all the extra cases below, but I think you're right and the extra verbosity is better than the computation.